### ISSUE TITLE
[a11y]: Popover storybook does not close on loss of focus, resulting in potential failure of 2.4.11

### ISSUE THREAD

**Issue by mbgower** [carbon-design-system/carbon#15917]
### Package

@carbon/react

### Browser

Chrome

### Package version

https://react.carbondesignsystem.com/?path=/story/components-popover--tab-tip#storybook-preview-wrapper

### Automated testing tool and ruleset

n/a

### Description

In storybook, most of the popover examples are (improperly) expanded by default, instead of being able to be triggered by the target. These should all be addressed.
However, there is one example, the tab tip, which provides both an expanded and unexpanded popover: https://react.carbondesignsystem.com/?path=/story/components-popover--tab-tip#storybook-preview-wrapper

The second one can be opened by keyboard and closes when it loses focus (as intended).
However, the first popover, which is already opened by default, must be manually closed by keyboard; if the user simply tabs out of the last item in the popover, it does not collapse, which it must do by default, or be at risk of failing [WCAG 2.2 Focus Not Obscured](https://www.w3.org/WAI/WCAG22/Understanding/focus-not-obscured-minimum.html).
Note that the second popover, which correctly collapses on loss of focus when tabbing, fails to close on loss of focus when shift+tabbing.

### WCAG 2.1 Violation

WCAG 2.2 Focus Not Obscured (Minimum)

### Reproduction/example

https://react.carbondesignsystem.com/?path=/story/components-popover--tab-tip#storybook-preview-wrapper

### Steps to reproduce

1. Launch https://react.carbondesignsystem.com/?path=/story/components-popover--tab-tip#storybook-preview-wrapper
2. use Space to activate the Tab Tip in the left navigation, then press Tab and activate Skip to Canvas, and navigate to the main region
3. Tab through the open Pop over and continuing tabbing through its children until the focus goes to the next gear icon (the first pop over should close on loss of focus; it does not)
4. Activate the second gear; the popover appears
5. Tab through all its children until focus leaves the popover; the popover collapses (as expected)
6. It is possible to also demonstrate a failure on the second popover. Shift+tab back to the second gear and press Space to open; the popover opens
7. Tab once or twice into the popover children, then Shift+Tab until the focus reaches the gear icon, and then one more time so it goes back to the first gear icon. The second popover should close at this point, but fails to do so.

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://github.com/carbon-design-system/carbon/blob/f555616971a03fd454c0f4daea184adf41fff05b/.github/CODE_OF_CONDUCT.md)
- [ ] I checked the [current issues](https://github.com/carbon-design-system/carbon/issues) for duplicate problems
